packages/abi-typegen/src/utils/getFunctionInputs.ts

```diff
@@ -0,0 +1,38 @@
+import { EmptyType } from '../abi/types/EmptyType';
+import { OptionType } from '../abi/types/OptionType';
+import type { IType } from '../types/interfaces/IType';
+import type { JsonAbiArgument } from '../types/interfaces/JsonAbi';
+
+import { findType } from './findType';
+
+export type ArgumentWithMetadata<TArg extends JsonAbiArgument = JsonAbiArgument> = TArg & {
+  isOptional: boolean;
+};
+
+export const getFunctionInputs = (params: {
```

Comment:
Same applies here , but why do we need to [redefine](https://github.com/FuelLabs/fuels-ts/pull/2777/files#diff-07695b2096aef52c7df838b7bdf2485bc032260a1d529ee5cd3fbff48e00c675R10) this exact same function?

Reply:
71a1309b87050e391251fef84fe2cd57ded1afe6

I redefined a similar function (with different inputs) due to the nature of two packages that are very closely linked `abi-coder` and `abi-typegen`. Once we merge these two packages (https://github.com/FuelLabs/fuels-ts/issues/2346), these two methods would sit together.